ui/src/buckets/components/UpdateBucketOverlay.tsx

```diff
@@ -35,31 +36,34 @@ import {OwnBucket} from 'src/types'
 
 interface DispatchProps {
   onUpdateBucket: typeof updateBucket
-  onNotify: typeof notify
 }
 
 type ReduxProps = ConnectedProps<typeof connector>
 type Props = ReduxProps & RouteComponentProps<{bucketID: string; orgID: string}>
 
 const UpdateBucketOverlay: FunctionComponent<Props> = ({
   onUpdateBucket,
-  onNotify,
   match,
   history,
 }) => {
   const {orgID, bucketID} = match.params
+  const dispatch = useDispatch()
   const [bucketDraft, setBucketDraft] = useState<OwnBucket>(null)
 
   const [loadingStatus, setLoadingStatus] = useState(RemoteDataState.Loading)
 
   const [retentionSelection, setRetentionSelection] = useState(DEFAULT_SECONDS)
 
+  const handleClose = useCallback(() => {
+    history.push(`/orgs/${orgID}/load-data/buckets`)
+  }, [orgID, history])
+
   useEffect(() => {
     const fetchBucket = async () => {
       const resp = await api.getBucket({bucketID})
```

Comment:
all this stuff is obviously in the wrong place {view layer instead of a data layer}, and while it's completely unrelated to the PR, it does beg the question; should we still use thunks if we have access to `useDispatch`?

Reply:
maybe i answered my own question.. move it to the data layer, and then we dont have useDispatch